### ISSUE TITLE
A query that asking interface doesn't return result through Apollo Gateway

### ISSUE THREAD

**Issue by rkudryashov** [async-graphql/async-graphql#95]
I have the following structure:

```
#[derive(Clone)]
pub struct SomeType {
    pub id: ID,
    test_field: CustomDecimal,
    test_field_2: Interface,
}

#[Object]
impl SomeType 

    ...

    async fn test_field_2(&self) -> &Interface {
        &self.test_field_2
    }
}
```

Let's say the interface has just one successor:

```
#[Interface(
field(name = "some_field", type = "i32", context),
)]
#[derive(Clone)]
enum Interface {
    Successor1(Successor1),
}

#[SimpleObject]
#[derive(Clone)]
struct Successor1 {
    some_field: i32,
}

```

Then I'm trying to request this field on `http://localhost:8001/`:

```
{
  someQuery {
    testField2 {
      someField
    }
  }
}
```

Result:

```
{
  "data": {
    "someQuery": {
      "testField2": {
        "someField": 3
      }
    }
  }
}
```
But if I perform the query on Apollo Gateway, it will fail:

```
{
  "data": null
}
```

Repro-project is here: https://github.com/rkudryashov/async-graphql-reproducer

**Comment by sunli829** [comment]
I'm still looking for reasons. 😁

**Comment by sunli829** [comment]
`async-graphql` did not provide `Federation` with the object interface information, I have fixed it in version `1.12.13`. 😁

**Comment by rkudryashov** [comment]
Nice work, thanks!
